src/main/resources/assets/admin/common/js/form/FormContext.ts

```diff
@@ -14,14 +17,15 @@ export class FormContext {
 
     private validationErrors: ValidationError[];
 
-    private readonly aiEditable: boolean;
+    private readonly aiTools: AiTool[];
```

Comment:
Think it's time we start using `Set`. It will result in more readable and simple code in our case, since we add unique values and try to find them later with `Array.some`